evalml/automl/automl_search.py

```diff
@@ -281,6 +285,13 @@ def __init__(
             objective = get_default_primary_search_objective(self.problem_type.value)
         objective = get_objective(objective, return_instance=False)
         self.objective = self._validate_objective(objective)
+        self.thresholding_objective = get_objective(
+            thresholding_objective, return_instance=True
+        )
+        if self.thresholding_objective.score_needs_proba:
+            raise ValueError(
+                "Thresholding objective must be thresholdable and cannot need probabilities!"
```

Comment:
This is a nit-pick but I don't think most people will understand what "thresholdable" means. Perhaps "Alternate threshold tuning objective must be a tuneable objective" ?